plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -131,7 +137,11 @@ func (l *persistentVolumeLabel) getEBSVolumes() (aws.Volumes, error) {
 	defer l.mutex.Unlock()
 
 	if l.ebsVolumes == nil {
-		cloudProvider, err := cloudprovider.GetCloudProvider("aws", nil)
+		var cloudConfigReader io.Reader
+		if len(l.cloudConfig) > 0 {
+			cloudConfigReader = bytes.NewReader(l.cloudConfig)
+		}
+		cloudProvider, err := cloudprovider.GetCloudProvider("aws", cloudConfigReader)
```

Comment:
Ideally the mechanism by which to read/write the config file should be in a single place (`cloudprovider` lib makes the most sense) instead of duplicated and sprinkled throughout the code.